src/migration/PoC_dfusion.js

```diff
@@ -31,6 +31,10 @@ async function migrate({ artifacts, deployer, network, accounts, web3, maxTokens
     "@gnosis.pm/solidity-data-structures/build/contracts/IterableAppendOnlySet"
   )
 
+  // Hack to populate truffle artifact data correctly for linked libraries.
+  await BiMap.deployed()
+  await IterableAppendOnlySet.deployed()
```

Comment:
Without first calling `.deployed()` on libraries which are linked to other contracts about to be deployed, I see  deployments fail with this error:
```
Error: 
    at Object.link (/Users/cgewecke/code/sc-forks/dex-contracts/node_modules/truffle/build/webpack:/packages/deployer/src/linker.js:29:1)
    at process._tickCallback (internal/process/next_tick.js:68:7)
Truffle v5.1.6 (core: 5.1.6)
Node v10.15.3
```
Honestly don't know why this is necessary or why it's surfacing in the plugin version of the tool. Feels like a bug within Truffle migrations - perhaps the artifact object has properties managed by getters/setters that aren't populated correctly unless `get` is called first in some contexts.  Idk.

It may also be specific to artifacts stored in node_modules - haven't seen it before and will keep an eye out.